src/components/PeopleFilters.tsx

```diff
@@ -1,27 +1,97 @@
-export const PeopleFilters = () => {
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import { Status } from '../types/status';
+import classNames from 'classnames';
+import { SearchLink } from './SearchLink';
+
+type Param = string | number;
+type Params = {
+  [key: string]: Param[] | Param | null;
+};
+
+type Props = {
+  handleFilter: () => Person[];
+};
+
+function getSearchWith(params: Params, search?: string | URLSearchParams) {
+  const newParams = new URLSearchParams(search);
+
+  for (const [key, value] of Object.entries(params)) {
+    if (value === null) {
+      newParams.delete(key);
+    } else if (value !== Status.All) {
+      newParams.delete(key);
+      newParams.set(key, value.toString());
+    } else if (Array.isArray(value)) {
+      newParams.delete(key);
+      value.forEach(item => {
+        return newParams.append(key, item.toString());
+      });
+    } else {
+      newParams.set(key, value.toString());
+    }
+  }
+
+  return newParams.toString();
+}
+
+export const PeopleFilters: React.FC<Props> = ({ handleFilter }) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const ages = ['16', '17', '18', '19', '20'];
+
+  function setSearchWith(params: any) {
+    const search = getSearchWith(params, searchParams);
+
+    setSearchParams(search);
+    handleFilter();
+  }
+
+  const handleSexStatus = (value: Status) => {
+    setSearchWith({ sex: value || null });
+  };
+
+  const handleQuerySearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchWith({ query: event.target.value || null });
+  };
+
+  const hasAgeInCenturies = (age: string) => {
+    return centuries.includes(age)
+      ? centuries.filter(ch => ch !== age)
```

Comment:
```suggestion
      ? centuries.filter(ch => ch !== age)
```

let's use more informative naming `ch => century`